velox/exec/Window.cpp

```diff
@@ -284,8 +284,25 @@ void updateKRowsOffsetsColumn(
   // moves ahead.
   int precedingFactor = isKPreceding ? -1 : 1;
   for (auto i = 0; i < numRows; i++) {
-    rawFrameBounds[i] =
-        (startRow + i) + vector_size_t(precedingFactor * offsets[i]);
+    auto startValue = (int64_t)(startRow + i) + precedingFactor * offsets[i];
```

Comment:
Nit : 

We can simplify the code a bit

```
rawFrameBounds[i] = (int64_t)(startRow + i) + precedingFactor * offsets[i];
if (rawFrameBounds[i] < INT32_MIN) {
  rawFrameBounds[i] = 0;
} else if (rawFrameBounds[i] > INT32_MAX) {
     // computeValidFrames will replace INT32_MAX set here
     // with partition's final row index.
  rawFrameBounds[i] = INT32_MAX;
}
```

Reply:
Hi @aditi-pandit, thanks for your review!
I just found `rawFrameBounds[i]` is int32. So maybe, we have to use `startValue` (int64) to store intermediate result, then do the checking. Right?